Sources/OpenAPIServiceModel/SetOperationInput.swift

```diff
@@ -0,0 +1,124 @@
+// Copyright 2019-2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License").
+// You may not use this file except in compliance with the License.
+// A copy of the License is located at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+//
+// SetOperationInput.swift
+// OpenAPIServiceModel
+//
+
+import Foundation
+import OpenAPIKit
+import ServiceModelEntities
+import ServiceModelCodeGeneration
+import Yams
+
+internal extension OpenAPIServiceModel {
+    struct OperationInputFields {
+        let allMembers: [String: Member]
+        let pathFields: [String]
+        let queryFields: [String]
+        let bodyFields: [String]
+        let additionalHeaderFields: [String]
+    }
+    
+    static func setOperationInput(bodyStructureName: String?,
+                                  operationInputMembers: OperationInputMembers,
+                                  description: inout OperationDescription,
+                                  model: inout OpenAPIServiceModel,
+                                  operationName: String) {
+        if let bodyStructureName = bodyStructureName,
+            operationInputMembers.queryMembers.isEmpty && operationInputMembers.additionalHeaderMembers.isEmpty
+                && operationInputMembers.pathMembers.isEmpty {
+            description.input = bodyStructureName
+        } else {
+            let operationInputFields = getOperationInputFields(bodyStructureName: bodyStructureName,
+                                                               operationInputMembers: operationInputMembers,
+                                                               model: &model)
+            
+            let inputModelName = "\(operationName)Request"
+            let inputModelDescription = "Input model for the \(operationName) operation."
+            let structureDefinition = StructureDescription(members: operationInputFields.allMembers,
+                                                           documentation: inputModelDescription)
+            
+            model.structureDescriptions[inputModelName] = structureDefinition
+            description.input = inputModelName
+            
+            if operationInputFields.pathFields.isEmpty && !operationInputFields.queryFields.isEmpty
+                && operationInputFields.bodyFields.isEmpty && operationInputFields.additionalHeaderFields.isEmpty {
+                description.inputDescription = OperationInputDescription(
+                    pathFields: operationInputFields.pathFields,
+                    queryFields: [],
```

Comment:
Why are we passing in an empty list here?  It seems like the if check specifies that the `operationInputFields.queryFields` is not empty.